### DISCUSSION TITLE
Sprinkler droplets traverse ceiling after plume forces ceiling impingement

### DISCUSSION THREAD

**Discussion by johodges** [firemodels/fds#12959] [General]
In large industrial fires such as rack storage systems, it is possible for sprinkler droplets to be overpowered by the plume and get carried away from the main fire. FDS can reproduce this behavior, but the results look a bit odd once the droplets impact the ceiling where the droplets start to move horizontally in a random direction along the ceiling. In reality the water droplets would probably impact something on the surface and re-enter the gas phase. It would be easy enough to include single cell thick obstructions on the ceiling which would then result in the droplets reaching the edge and dropping again. Curious on the team's thoughts on this topic.

https://github.com/firemodels/fds/assets/17127980/edf784de-a22d-432b-8b09-7b0457ab2c04

**Comment by mcgratta** [comment]
There is a parameter called `ALLOW_UNDERSIDE_PARTICLES` which says that a droplet can move along a downward-facing surface. This was put in a long time ago to deal with rack storage cases where the underside droplets represented all the water that penetrated the stack through many different means, none of which are directly modeled. This is not a global parameter, but it could be moved to the `SURF` line to avoid what you are seeing.

**Reply by Snickkkk** [reply]
Hi there, I'm trying to run the rack storage output file on my computer but I keep getting an error. The input file seems not operational. Could you help, please?

**Reply by mcgratta** [reply]
What is the error message? What version of FDS are you using? Do you have an input file that you can share? If you have this information, start a new issue. This one is closed.

**Reply by drjfloyd** [reply]
This is not nearly enough information to help. Start a new discussion and be very specific. 

Is this a file from the FDS repository? If so exactly which one.
Is this your own file? If so we will probably need to see the file.
Are you running the current release of FDS? If not, do so.
If you are running a file from the FDS repository is it from the same version of the repository as the FDS release you are using?
Are you following the exact instructions given in the FDS User's Guide on how to run FDS?
What specific error message are you seeing?

**Reply by Snickkkk** [reply]
Thanks very much for your quick answer.
Is this a file from the FDS repository? If so exactly which one.? Here is the file: https://github.com/firemodels/fds/blob/master/Validation/UL_NFPRF/FDS_Input_Files/UL_NFPRF_P-5.fds
Is this your own file? If so we will probably need to see the file. No it's not mine. I downloaded it from the link above
Are you running the current release of FDS? If not, do so.? I'm running FDS6 which is the current version
If you are running a file from the FDS repository is it from the same version of the repository as the FDS release you are using? That I don't know. How can I figure this out?
Are you following the exact instructions given in the FDS User's Guide on how to run FDS? I guess I am
What specific error message are you seeing? Here are the errors:

C:\firemodels\FDS6>fds UL_NFPRF_P-5.fds
 Reading input file ...

ERROR: Problem with SURF number4, line number94 (CHID: UL_NFPRF_P-5)


 DEVICE Activation Times

 101  vent link 1                                                 No Activation
 102  vent link 2                                                 No Activation
 103  vent link 3                                                 No Activation
 104  vent link 4                                                 No Activation
 105  vent link 5                                                 No Activation


 CONTROL Activation Times

   1  Act01                                                       F No Activation
   2  Act02                                                       F No Activation
   3  Act03                                                       F No Activation
   4  Act04                                                       F No Activation
   5  Act05                                                       F No Activation
   6  Act06                                                       F No Activation
   7  Act07                                                       F No Activation
   8  Act08                                                       F No Activation
   9  Act09                                                       F No Activation
  10  Act10                                                       F No Activation
  11  Act11                                                       F No Activation
  12  Act12
Thanks very much for your help.

**Reply by mcgratta** [reply]
Type 
```
fds
```
and report what text follows.

**Reply by Snickkkk** [reply]
Sorry didnt get. What text follows?

**Reply by mcgratta** [reply]
![image](https://github.com/user-attachments/assets/3bb872ed-96d5-4df3-b490-4913b52d3943)

**Reply by Snickkkk** [reply]
Reading input file ...

 Fire Dynamics Simulator

 Current Date     : August  7, 2024  15:36:39
 Revision         : FDS6.7.0-0-g5ccea76-master
 Revision Date    : Mon Jun 25 13:03:23 2018 -0400
 Compiler         : Intel ifort 18.0.2.185
 Compilation Date : Tue 06/26/2018  10:49 AM

 MPI Enabled;    Number of MPI Processes:       1
 OpenMP Enabled; Number of OpenMP Threads:      4

 MPI version: 3.1
 MPI library version: Intel(R) MPI Library 2018 Update 2 for Windows* OS


 Consult FDS Users Guide Chapter, Running FDS, for further instructions.

 Hit Enter to Escape...

**Reply by Snickkkk** [reply]
Did you run the input file on your PC to see it works?

**Comment by mcgratta** [comment]
I moved `ALLOW_UNDERSIDE_PARTICLES` and `ALLOW_SURFACE_PARTICLES` from `MISC` to `SURF`. This gives us better control over these.

**Comment by Snickkkk** [comment]
Thanks very much for your quick answer.
Is this a file from the FDS repository? If so exactly which one.? Here is
the file:
https://github.com/firemodels/fds/blob/master/Validation/UL_NFPRF/FDS_Input_Files/UL_NFPRF_P-5.fds
Is this your own file? If so we will probably need to see the file. No it's
not mine. I downloaded it from the link above
Are you running the current release of FDS? If not, do so.? I'm running
FDS6 which is the current version
If you are running a file from the FDS repository is it from the same
version of the repository as the FDS release you are using? That I don't
know. How can I figure this out?
Are you following the exact instructions given in the FDS User's Guide on
how to run FDS? I guess I am
What specific error message are you seeing? Here are the errors:
[image: image.png]
Thanks very much for your help.
Nick

**Comment by Snickkkk** [comment]
Kevin,
This is what I get:

[image: image.png]
Thanks a lot for your help

**Comment by mcgratta** [comment]
You are using a version of FDS from 2018, but you have also taken an input file from the most recent version of FDS. Your old FDS and your new input file are not compatible. If you want to run the old version of FDS, then clone the github repository and use a version of the input file that is consistent with the version of FDS>

**Reply by Snickkkk** [reply]
Thanks for letting me know. Or I can download the most recent version of fds and run the input file, correct? I don't have the 2018 version of the inut file otherwise I'd use that.

**Reply by mcgratta** [reply]
If you clone the GitHub fds repository, you can get the older version of the input file.

**Reply by Snickkkk** [reply]
I cloned the GitHub repository. Where should I go to find the older version of the input file? Thanks,

**Reply by drjfloyd** [reply]
It is in the Validaiton folder wherever you cloned the repo into. The path from there is the same as your github url from two days ago.

**Reply by Snickkkk** [reply]
When I go to the folder I just see the most recent version. How cloning the repository would allow me to have the older version of the input file? I've cloned it which is essentially copying the folder on my computer. That'd be great if you could help.

**Reply by drjfloyd** [reply]
A couple of options:

1) 
from a command prompt while in your repo
git checkout tag
where tag is the tag for whatever version you want.  The tag can be seen on the releases page. 

![image](https://github.com/user-attachments/assets/0ac2eeed-3405-48c6-af29-22b567ec2854)

2) 

On the releases web page, click on the tag for the version you want. That will set the website to that version. Then navigate to the fodler with the input file you want, select the file, and then you can click on the download button.

**Reply by Snickkkk** [reply]
Thanks for your note. I can follow what you mentioned. Would it be possible to call you? I can email you my phone number. Thanls

**Comment by Snickkkk** [comment]
Hi there,
I just wanted to thank you deeply for guiding me in running the model. I could run it successfully. Thanks again.
